### ISSUE TITLE
The terraform docs and examples are based on an old version of the cloudflare provider

### ISSUE THREAD

**Issue by cpick** [cloudflare/cloudflare-docs#4]
The tutorial shows examples from and is based on an old version of the cloudflare terraform provider.

For example, the [Hello World example](https://github.com/cloudflare/cloudflare-docs/blob/7d7c4f43b91c711e4be0aabefb17cbd71c620c51/products/terraform/src/content/tutorial/hello-world.md#1-defining-your-first-terraform-config-file) uses the `token` parameter while that's been [renamed to `api_key`](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/guides/version-2-upgrade#provider-global-configuration-changes).

A much more minor nitpick: there are several typos throughout the guide (eg: "As you configuration" on the Hello World guide).

Thanks for creating the Terraform provider and the `cf-terraforming` importer, they're very useful!  It was just a bit harder to get started than it should have been since the docs were out of date.

**Comment by vivaladan** [comment]
Terraform 0.13 requires a different style configuration block. It now needs to read

```
terraform {
  required_providers {
    cloudflare = {
      source = "terraform-providers/cloudflare"
    }
  }
}
```

Otherwise it will error with

> Error: Failed to install providers
> 
> Could not find required providers, but found possible alternatives:
> 
>   hashicorp/cloudflare -> cloudflare/cloudflare

**Comment by jacobbednarz** [comment]
These issues should be addressed by #397. Please drop a note in that issue if you see something that isn't covered here.

**Comment by KimJ15** [comment]
Closing this as resolved based on #397.
